lib/matplotlib/axes/_axes.py

```diff
@@ -3275,9 +3295,7 @@ def pie(self, x, explode=None, labels=None, colors=None,
         The Axes aspect ratio can be controlled with `.Axes.set_aspect`.
         """
         self.set_aspect('equal')
-        # The use of float32 is "historical", but can't be changed without
-        # regenerating the test baselines.
-        x = np.asarray(x, np.float32)
+        x = np.asarray(x)
```

Comment:
Casting to float here meant that we couldn't use "%d" for the format string, which is the most obvious thing to use if your input values are int.  No tests failed for me locally.  I note that tolerances were recently added for numpy v2:
https://github.com/matplotlib/matplotlib/blob/183b04fb43f57ffe66da68c729a179e397cd35f6/lib/matplotlib/tests/test_axes.py#L5947-L5950